subjects/gaming/firing-range/README.md

```diff
@@ -1,39 +1,125 @@
+
+- Create a stationary target.
+- Create a moving target with basic AI that follows a predefined path or pattern.
+- Implement hit detection for the targets.
+- Ensure the targets respawn after a set interval once they are hit.
+<center>
+  <div style="display: flex; justify-content: center;">
+    <img src="./resources/example1.jpg?raw=true" style="width: 500px !important; height: 350px !important;" />
+    <img src="./resources/example2.jpg?raw=true" style="width: 500px !important; height: 350px !important;" />
+  </div>
+</center>
 
-For this project you will have to create a Blueprint Class target, that will have some of these characteristics. The target should :
+#### Game Loop Logic
 
-- have associated to it the previous created material.
-- either be moving from side to side or be stationary.
-- be dynamic, using the timeline node.
-- use a public variable to set or unset the movement animation of the target.
-- rise again after x seconds after being hit it with a projectile, and behave like before it was hit.
+The game loop must be continuous and consist of:
 
-Only one class of target is allowed for the whole project.
+- The player starts with a loaded weapon and can shoot at targets.
+- The player can open a pause menu at any time, with options to:
+  - Restart the game (reset score, targets, and ammo).
+  - Return to the main menu.
 
-The previously mentioned projectile should:
+#### Level Design
 
-- have a size of X = Y = Z = 0,5.
-- have a speed of 10000.
+You must design a firing range that includes:
 
-After downloading and unzipping this [file](https://assets.01-edu.org/Unreal-Engine-Piscine/FiringRange.zip), you can copy its content to your project Content folder.
+- A coherent theme to ensure a consistent and immersive experience.
+- Proper lighting and environment setup to ensure the range is clear and visually appealing.
+- A section with stationary targets.
+- A section with moving targets.
+- Ammo pickups where the player can restock.
 
-When finished, your project should look like the executable example on the folder or the ["Expected Result" video](https://youtu.be/EBibaN-dh_0).
+### Bonus
 
-> Do not forget to zip up the project compile and save everything for peer correction.
-> If it is not possible to upload files to Gitea due to their size, use GitHub instead and have a look at [Git LSF](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github)
+- Add multiple weapons (e.g., a shotgun, sniper rifle) with unique shooting and reload mechanics.
+- Implement more advanced target AI, such as random movement patterns or different difficulty levels.
+- Add different hit zones for targets, with headshots awarding more accuracy points.
+- Design a timed challenge mode where the player must hit all targets within a set time frame.
 
-#### Bonus
+### Submission
 
-Here are some ideas for improving the game:
+- You must upload a zip file of your game build in your repository.
```

Comment:
Is it not better to have just the code then rebuild the project during the audit ?

Reply:
an unreal engine project is significantly larger than the build. the build only packages the necessary parts of the engine that were used in your game, and a build process can take quite long.

Reply:
Okay thank for the infos 🙏